### PR TITLE
Force first location update without last location

### PR DESCRIPTION
**Pull request by danesfeder**
Sometimes a given `LocationEngine` won't have a last known location (`LocationEngine#getLastLocation()`). 

Now, if this is the case, we create a location update from the first coordinate of the route to fire the `ProgressChangeListener` and populate a given UI.

### CODE REVIEW

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java [hunk 1]
```diff
@@ -284,7 +285,7 @@ private void startForegroundNotification(Notification notification, int notifica
    * performing navigation progress on a accurate/valid rawLocation update.
    */
   @SuppressWarnings("MissingPermission")
-  private boolean validLocationUpdate(Location location) {
+  private boolean isValidLocationUpdate(Location location) {
     if (locationEngine.getLastLocation() == null) {
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java]: I think this check should be removed.

If you 👀 the code carefully we're always checking if `location != null` before calling `isValidLocationUpdate` so we know for sure that the location is not `null` at that point. Even more, if the `location` received https://github.com/mapbox/mapbox-navigation-android/pull/756/files#diff-7e57c49bef82db11f00cfa982b5e2e07R99 is not `null` but the `locationEngine.getLastLocation() == null` we're not checking if the location is the same as previous. It that what we want?

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java [hunk 2]
```diff
@@ -301,8 +302,11 @@ private boolean validLocationUpdate(Location location) {
   @SuppressWarnings("MissingPermission")
   private void forceLocationUpdate() {
     Location lastLocation = locationEngine.getLastLocation();
-    if (lastLocation != null) {
-      thread.queueTask(MSG_LOCATION_UPDATED, NewLocationModel.create(lastLocation, mapboxNavigation));
+    if (lastLocation != null && isValidLocationUpdate(lastLocation)) {
+      onLocationChanged(lastLocation);
+    } else {
+      Location forcedLocation = RouteUtils.createFirstLocationFromRoute(mapboxNavigation.getRoute());
+      thread.queueTask(MSG_LOCATION_UPDATED, NewLocationModel.create(forcedLocation, mapboxNavigation));
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java]: The `if else` could be refactored calling to `onLocationChanged` instead.

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.java [hunk 3]
```diff
@@ -102,6 +105,26 @@ public static boolean isLastLeg(RouteProgress routeProgress) {
...
+   * This method creates a location update from the first coordinate (origin) that created
+   * the route.
+   *
+   * @param route with list of coordinates
+   * @return {@link Location} from first coordinate
+   * @since 0.10.0
+   */
+  public static Location createFirstLocationFromRoute(DirectionsRoute route) {
+    List<Point> coordinates = route.routeOptions().coordinates();
+    Point origin = coordinates.get(0);
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/RouteUtils.java]: Magic number.

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java (hunk 1 above)
Comment (devotaaabel) [reply libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java]: Yea I think this method should be removed. I think at some point this method was useful but I don't think it really is just for a null check, especially if it's redundant. If you look at the comments, it says it `runs several checks`.

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java [hunk 4]
```diff
@@ -284,26 +285,40 @@ private void startForegroundNotification(Notification notification, int notifica
...
    * At the very beginning of navigation session, a forced location update occurs so that the
    * developer can immediately get a routeProgress object to display information.
    */
   @SuppressWarnings("MissingPermission")
   private void forceLocationUpdate() {
+    Location forcedLocation;
     Location lastLocation = locationEngine.getLastLocation();
-    if (lastLocation != null) {
-      thread.queueTask(MSG_LOCATION_UPDATED, NewLocationModel.create(lastLocation, mapboxNavigation));
+    if (isValidLocationUpdate(lastLocation)) {
```
Comment (Guardiola31337) [inline libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java]: Minor refactor 👀
```java
Location location = locationEngine.getLastLocation();
if (!isValidLocationUpdate(lastLocation)) {
  location = RouteUtils.createFirstLocationFromRoute(mapboxNavigation.getRoute());
}
queueLocationUpdateTask(location);
```

Removing complexity 😉

**Pull request merged**